algorithms/src/Kokkos_Random.hpp

```diff
@@ -965,6 +964,40 @@ class Random_XorShift64_Pool {
     Kokkos::memory_fence();
     locks_(state.state_idx_, 0) = 0;
   }
+
+ private:
+  void init_members(int num_states) {
```

Comment:
You'd need to pass through the execution space here but I would just "inline" the definition.
It's only called once if you use a delegating constructor as suggested above.